### PR TITLE
BUG: axis kw not propogating on pct_change

### PR DESCRIPTION
**Pull request by Tux1**
axis option was not correctly set on fillna in pct_change

### CODE REVIEW

**Comment by sinhrks** [comment]
Thanks. Can you add a test for this?

**Comment by jreback** [comment]
@Tux1 pls add a test for this

**Comment by Tux1** [comment]
Sorry, it is my first time.
Is it okay ? I pushed a simple test that shows the issue and my previous commit fixed it.
Thanks

**Comment by Tux1** [comment]
Sorry. It is my first commit.
I made a simple test that shows the issue and my commit fixed it.
Is it okay ?
Thanks

**Comment by jreback** [comment]
pls see the contributing docs: http://pandas.pydata.org/pandas-docs/stable/contributing.html

tests / fix should be on a single pr.

**Comment by Tux1** [comment]
Sorry... Moreover the test was wrong. 
I have made another test which is passed with my patch. That is committed in a single branch, this one.

File: pandas/tests/test_generic.py [hunk 1]
```diff
@@ -1851,6 +1851,15 @@ def test_pipe_panel(self):
         with tm.assertRaises(ValueError):
             result = wp.pipe((f, 'y'), x=1, y=1)
 
+    def test_pct_change(self):
+        pnl = DataFrame([np.arange(0, 40, 10), np.arange(0, 40, 10), np.arange(0, 40, 10)]).astype(np.float64)
```
Comment (jreback) [inline pandas/tests/test_generic.py]: add the issue number here

**Comment by jreback** [comment]
pls add a whatsnew note for 0.17.1 (bug fix)

File: pandas/tests/test_generic.py [hunk 2]
```diff
@@ -1851,6 +1851,15 @@ def test_pipe_panel(self):
...
+        pnl.iat[1,0] = np.nan
+        pnl.iat[1,1] = np.nan
+        
+        expected = pnl.ffill(axis=1).pct_change(axis=1, fill_method=None)
```
Comment (jreback) [inline pandas/tests/test_generic.py]: test with axis 0, 2 as well (if their are existing tests for that, pls move them here)

further this should be a bit more generic (or append _frame) to the test name as the ``generic.py`` tests should test Series,DataFrame,Panel

**Comment by jreback** [comment]
@Tux1 can you rebase / update

**Comment by Tux1** [comment]
Sorry for several commits. I think that is okay now ? whatsnew edited, commit rebased

File: doc/source/whatsnew/v0.17.1.txt [hunk 3]
```diff
@@ -96,7 +96,7 @@ Bug Fixes
 
 
 
-
+- Bug in ``DataFrame.pct_change()`` was not propagating axis on fillna method (:issue:`11150`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.17.1.txt]: use double backticks around ``axis`` (and say ``axis`` keyword) and 'on ``.fillna()`` method'

File: pandas/tests/test_generic.py [hunk 4]
```diff
@@ -1859,6 +1859,26 @@ def test_pipe_panel(self):
         with tm.assertRaises(ValueError):
             result = wp.pipe((f, 'y'), x=1, y=1)
 
+    def test_pct_change_frame(self):
+        pnl = DataFrame([np.arange(0, 40, 10), np.arange(0, 40, 10), np.arange(0, 40, 10)]).astype(np.float64)
```
Comment (jreback) [inline pandas/tests/test_generic.py]: add the issue number as a comment here

File: pandas/tests/test_generic.py [hunk 5]
```diff
@@ -1859,6 +1859,26 @@ def test_pipe_panel(self):
...
+        pnl.iat[2,3] = 60
+        
+        mask = pnl.isnull()
+
+        expected_axis0 = pnl.ffill(axis=0)/pnl.ffill(axis=0).shift(axis=0) - 1
+        expected_axis0[mask] = np.nan
+        result_axis0 = pnl.pct_change(axis=0, fill_method='pad')
+
+        self.assert_frame_equal(result_axis0, expected_axis0)
+
```
Comment (jreback) [inline pandas/tests/test_generic.py]: move this to ``test_frame`` right after the other ``test_pct_change_*`` tests

File: pandas/tests/test_generic.py [hunk 6]
```diff
@@ -1859,6 +1859,26 @@ def test_pipe_panel(self):
...
 
+    def test_pct_change_frame(self):
+        pnl = DataFrame([np.arange(0, 40, 10), np.arange(0, 40, 10), np.arange(0, 40, 10)]).astype(np.float64)
+        pnl.iat[1,0] = np.nan
+        pnl.iat[1,1] = np.nan
+        pnl.iat[2,3] = 60
+        
+        mask = pnl.isnull()
+
+        expected_axis0 = pnl.ffill(axis=0)/pnl.ffill(axis=0).shift(axis=0) - 1
```
Comment (jreback) [inline pandas/tests/test_generic.py]: you can do this in a loop over both axes

**Comment by jreback** [comment]
some comments, pls squash when finished

File: pandas/tests/test_generic.py (hunk 5 above)
Comment (Tux1) [inline pandas/tests/test_generic.py]: what do you mean ? There is no ``test_frame`` and I made the only one ``test_pct_change_frame`` test

**Comment by jreback** [comment]
pls rebase on master / squash, see docs [here](http://pandas.pydata.org/pandas-docs/stable/contributing.html#contributing-your-changes-to-pandas)

File: pandas/tests/test_generic.py [hunk 7]
```diff
@@ -1859,6 +1859,23 @@ def test_pipe_panel(self):
         with tm.assertRaises(ValueError):
             result = wp.pipe((f, 'y'), x=1, y=1)
 
+    # GH 11150
+    def test_pct_change_frame(self):
+        pnl = DataFrame([np.arange(0, 40, 10), np.arange(0, 40, 10), np.arange(0, 40, 10)]).astype(np.float64)
+        pnl.iat[1,0] = np.nan
+        pnl.iat[1,1] = np.nan
```
Comment (jreback) [inline pandas/tests/test_generic.py]: I mean that this function needs to move to ``tests\test_frame.py`` (and not in ``test_generic.py``)

**Comment by Tux1** [comment]
Sorry, my repo is a mess... I will be better next time

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/334b07641a1dc8c1e7390d01184e114ddde0a7da

thanks!

**Pull request closed without merging**